core/math/vector2.cpp

```diff
@@ -35,17 +35,16 @@ real_t Vector2::angle() const {
 }
 
 real_t Vector2::length() const {
-	return Math::sqrt(x * x + y * y);
+	return Math::sqrt(length_squared());
 }
 
 real_t Vector2::length_squared() const {
 	return x * x + y * y;
 }
 
 void Vector2::normalize() {
-	real_t l = x * x + y * y;
+	real_t l = length();
```

Comment:
It was done this way to avoid calculating the square root needlessly in the case where `l == 0`... but I guess `sqrt(0)` should be cheap enough, so it might be fine.

But `length()` and `length_squared()` should maybe be `__FORCE_INLINE__`-ed.

Reply:
There are still many methods doesn't simplified, for example like Vector3